pkg/machines/libvirt-dbus.js

```diff
@@ -800,11 +804,55 @@ const LIBVIRT_DBUS_PROVIDER = {
                             dispatch(updateVm(Object.assign({}, props, dumpxmlParams)));
                         else
                             dispatch(updateOrAddVm(Object.assign({}, props, dumpxmlParams)));
+
+                        dispatch(getVmSnapshots({ connectionName, domainPath: objPath }));
                     })
                     .catch(ex => console.warn("GET_VM action failed for path", objPath, ex.toString()));
         };
     },
 
+    GET_DOMAIN_SNAPSHOTS({ connectionName, domainPath }) {
+        return dispatch => call(connectionName, domainPath, 'org.libvirt.Domain', 'ListDomainSnapshots', [0], { timeout, type: 'u' })
+                .then(objPaths => {
+                    const snaps = [];
+                    const promises = [];
+
+                    objPaths[0].forEach(objPath => promises.push(call(connectionName, objPath, 'org.libvirt.DomainSnapshot', 'GetXMLDesc', [0], { timeout, type: 'u' })));
+
+                    // WA to avoid Promise.all() fail-fast behavior
+                    const toResultObject = (promise) => {
+                        return promise
+                                .then(result => ({ success: true, result }))
+                                .catch(error => ({ success: false, error }));
+                    };
+
+                    Promise.all(promises.map(toResultObject))
+                            .catch(ex => console.warn("DomainSnapshot method GetXMLDesc failed", ex.toString()))
+                            .then(snapXmlList => {
+                                snapXmlList.forEach(snap => {
+                                    if (snap.success) {
+                                        const snapXml = snap.result[0];
+                                        const dumpxmlParams = parseDomainSnapshotDumpxml(snapXml);
+                                        snaps.push(dumpxmlParams);
+                                    }
```

Comment:
No "else" here to handle the `error` that toResultObject() returned?

Reply:
Not really I guess. If it returns error, I guess the best is to just ignore that specific domain snapshot and
Hmm, when I look at it now id kinda doesn't make sense. If `toResultObject` fails, is it or is it not caught by
`.catch(ex => console.warn("DomainSnapshot method GetXMLDesc failed", ex.toString()))`
?
If it is caught, there is no reason to test for if (snap.success). If it is not, then that .catch(...) there is redundant and I should add else with some error handling,

Reply:
Got rid of the .catch and added else which handles errors.